evalml/pipelines/components/transformers/preprocessing/polynomial_decomposer.py

```diff
@@ -0,0 +1,248 @@
+
+        Raises:
+            ValueError: If y is None.
+        """
+        if y is None:
+            raise ValueError("y cannot be None for PolynomialDecomposer!")
+        y_ww = infer_feature_types(y)
+
+        # Add polynomial trend back to signal
+        y_retrended = self._component_obj.inverse_transform(y_ww)
+
+        # Determine where the seasonality starts
+        first_index_diff = y_ww.index[0] - self.seasonality.index[0]
+        # TODO: Write tests to test different time series frequencies.
+        if self.frequency == "D":
+            delta = timedelta(days=1)
+            period = timedelta(days=self.periodicity)
+
+        # Cycle through the saved seasonality to match the first index of the transform data and project forward to the last index
+        transform_first_ind = int((first_index_diff % period) / delta)
+        seasonal = np.tile(
+            np.roll(self.seasonality.T.values, -transform_first_ind),
+            len(y_ww) // self.periodicity + 1,
+        ).T[: len(y_ww)]
+
+        y_t = infer_feature_types(pd.Series(y_retrended + seasonal, index=y_ww.index))
+        return y_t
+
+    def get_trend_dataframe(self, X, y):
+        """Return a list of dataframes with 3 columns: trend, seasonality, residual.
+
+        Scikit-learn's PolynomialForecaster is used to generate the trend portion of the target data. statsmodel's
+        seasonal_decompose is used to generate the seasonality of the data.
+
+        Args:
+            X (pd.DataFrame): Input data with time series data in index.
+            y (pd.Series or pd.DataFrame): Target variable data provided as a Series for univariate problems or
+                a DataFrame for multivariate problems.
+
+        Returns:
+            list of pd.DataFrame: Each DataFrame contains the columns "trend", "seasonality" and "residual,"
+                with the column values being the decomposed elements of the target data.
+
+        Raises:
+            TypeError: If X does not have time-series data in the index.
+            ValueError: If time series index of X does not have an inferred frequency.
+            TypeError: If y is not provided as a pandas Series or DataFrame.
+
+        """
+        X = infer_feature_types(X)
+        if not isinstance(X.index, pd.DatetimeIndex):
+            raise TypeError("Provided X should have datetimes in the index.")
+        if X.index.freq is None:
+            raise ValueError(
+                "Provided DatetimeIndex of X should have an inferred frequency."
+            )
+        fh = ForecastingHorizon(X.index, is_relative=False)
+
+        result_dfs = []
+
+        def _decompose_target(X, y, fh):
+            """Function to generate a single DataFrame with trend, seasonality and residual components."""
+            forecaster = self._component_obj.forecaster.clone()
+            forecaster.fit(y=y, X=X)
+            trend = forecaster.predict(fh=fh, X=y)
```

Comment:
Having been burned by all the partial dependence work, I'm wondering why we need to clone and retrain here. Would it be simpler just to call `trend = self._component_obj.forecaster.predict(...)`? Or is there something I'm missing.

Reply:
Respect.  I think what I was trying to get at here was that I didn't want to underlying transformer to be refit on other data and possibly "mess it up".  Is that at all a valid concern?

Reply:
Oh I totally agree we shouldn't refit without cloning the component object - that could definitely be problematic. I'm thinking more just calling predict, requiring the component to be fit before this. I don't think that would cause too many problems - if someone wanted to use different data, they'd probably want to use a different instance of this component anyway.

Reply:
Good call, I'll make the changes.